sbi/inference/snle/snle_base.py

```diff
@@ -116,104 +142,34 @@ def __call__(
                 samples.
             retrain_from_scratch_each_round: Whether to retrain the conditional density
                 estimator for the posterior from scratch each round.
+            show_train_summary: Whether to print the number of epochs and validation
+                loss after the training.
 
         Returns:
-            Posterior $p(\theta|x_o)$ that can be sampled and evaluated.
+            Density estimator that has learned the distribution $p(x|\theta)$.
         """
 
         max_num_epochs = 2 ** 31 - 1 if max_num_epochs is None else max_num_epochs
 
-        self._check_proposal(proposal)
-        self._round = self._round + 1 if (proposal is not None) else 0
-
-        # If presimulated data was provided from a later round, set the self._round to
-        # this value. Otherwise, we would rely on the user to _additionally_ provide the
-        # proposal that the presimulated data was sampled from in order for self._round
-        # to become larger than 0.
-        if self._data_round_index:
-            self._round = max(self._round, max(self._data_round_index))
-
-        # Run simulations for the round.
-        theta, x = self._run_simulations(proposal, num_simulations)
-        self._append_to_data_bank(theta, x, self._round)
-
         # Load data from most recent round.
-        theta, x, _ = self._get_from_data_bank(self._round, exclude_invalid_x, False)
+        self._round = max(self._data_round_index)
+        theta, x, _ = self.get_data(self._round, exclude_invalid_x, False)
 
         # First round or if retraining from scratch:
         # Call the `self._build_neural_net` with the rounds' thetas and xs as
         # arguments, which will build the neural network
         # This is passed into NeuralPosterior, to create a neural posterior which
         # can `sample()` and `log_prob()`. The network is accessible via `.net`.
-        if self._posterior is None or retrain_from_scratch_each_round:
-            x_shape = x_shape_from_simulation(x)
+        if self._neural_net is None or retrain_from_scratch_each_round:
```

Comment:
I see, the posterior remains `None` until after training, until `build_posterior` is called.

Reply:
Yes!